ivadomed/loader/utils.py

```diff
@@ -165,35 +165,26 @@ def imed_collate(batch):
     return batch
 
 
-def filter_roi(ds, nb_nonzero_thr):
+def filter_roi(roi_data, nb_nonzero_thr):
     """Filter slices from dataset using ROI data.
 
-    This function loops across the dataset (ds) and discards slices where the number of
-    non-zero voxels within the ROI slice (e.g. centerline, SC segmentation) is inferior or
-    equal to a given threshold (nb_nonzero_thr).
+    This function filters slices (roi_data) where the number of non-zero voxels within the ROI slice (e.g. centerline,
+    SC segmentation) is inferior or equal to a given threshold (nb_nonzero_thr).
 
     Args:
-        ds (mt_datasets.MRI2DSegmentationDataset): Dataset.
+        roi_data (nd.array): ROI slice.
         nb_nonzero_thr (int): Threshold.
 
     Returns:
-        mt_datasets.MRI2DSegmentationDataset: Dataset without filtered slices.
+        bool: True if the slice needs to be filtered, False otherwise.
     """
-    filter_indexes = []
-    for segpair, slice_roi_pair in ds.indexes:
-        roi_data = slice_roi_pair['gt']
+    # Discard slices with less nonzero voxels than nb_nonzero_thr
+    if not np.any(roi_data):
+        return True
+    if np.count_nonzero(roi_data) <= nb_nonzero_thr:
+        return True
 
-        # Discard slices with less nonzero voxels than nb_nonzero_thr
-        if not np.any(roi_data):
-            continue
-        if np.count_nonzero(roi_data) <= nb_nonzero_thr:
-            continue
-
-        filter_indexes.append((segpair, slice_roi_pair))
-
-    # Update dataset
-    ds.indexes = filter_indexes
-    return ds
+    return False
```

Comment:
Suggestion:
```python
return not np.any(roi_data) or np.count_nonzero(roi_data) <= nb_nonzero_thr
```
 instead of 
```python
    if not np.any(roi_data):
        return True
    if np.count_nonzero(roi_data) <= nb_nonzero_thr:
        return True

    return False
```

Reply:
brilliant